pages/api/account/manage/event/[[...data]].js

```diff
@@ -21,6 +25,15 @@ export default async function handler(req, res) {
       .json({ error: "Invalid request: GET or PUT or DELETE required" });
   }
 
+  try {
+    const account = await getAccountByProviderAccountId(session.user.id);
+    if (account) {
+      userId = account._id
+    }
+  } catch (error) {
+    logger.error(error, `Changelog - failed to get account for username: ${username}`);
+  }
```

Comment:
Not all requests come via http, they can also go direct to the functions below. I think this needs moving into each function?

Reply:
D'you mean I should run the `getAccountByProviderUserId` to populate the `userId` separately within each endpoint function!? 🤔

Reply:
Yes, because some of these functions are called directly within the app - you could add a new function in the file that is called in each function

Reply:
Oh, this means that the `called directly` do not go through the controller/handler function in each endpoint... 🤔 

```js
export default async function handler(req, res) {
  ....
}
```

...correct @eddiejaoude?

Reply:
yes correct 👍  this can happen with nextjs as it is recommend not to go via http (the handler) for server side calls

Reply:
Oh, that's true!

But, my reservation now would be this! will be able to retrieve the `session` object without the a `context` object!? 🤔 

```js
const session = await getServerSession(req, res, authOptions)
```

Put simple, I will require the `session.user.id` property...

Maybe i cannot use the `getServerSession` but I can use the `getSession` but is that not meant for client-side? Plus I understand if it will be used on the server then...

> NOTE: When calling getSession() server side, you need to pass {req} or context object. 
https://next-auth.js.org/v3/getting-started/client#getsession

Any ideas!?

Reply:
We had this challenge somewhere else and came up with a solution, let me try to find it in the project

Reply:
Oh great resource!

I'll look to implement with this style.

Reply:
Thank you 👍  also let us know if this can be improved